slsa/v1alpha1/provenance.proto

```diff
@@ -90,3 +92,16 @@ message Provenance {
   // Materials is a collection of artifacts used in the build.
   repeated Material materials = 4;
 }
+
+// A ProvenanceStatement ties together a subject, a provenance, and an attestation.
+message ProvenanceStatement {
```

Comment:
I decided to create a specific `ProvenanceStatement` type instead of having a `Statement` with a generic predicate, as that was difficult to model. Originally I tried to use `google.protobuf.Struct`, which maps to a `map[string]interface{}` in Go, but converting a `Provenance` into the `structpb.Struct` type involves serializing/deserializing or a lib like [mapstructure](https://github.com/mitchellh/mapstructure) (which I don't think would work here without adding the struct tags to the message).

I also looked at `google.protobuf.Any`. This is for encoding arbitrary messages, so it seemed like a better fit. However, it adds a `@type` field to the message to help with deserializing, which means that the JSON would differ from the spec. It also adds some complexity to working with the protos, since the predicate would
have to be wrapped by `anypb.New()` and then unwrapped with `anypb.UnmarshalTo()`.

Reply:
I suppose another alternative would be to make the `Statement` generic, but then use `oneof` for the predicate.  Is that an option too, or am I missing something here?

Reply:
I guess I wasn't the biggest fan of the way that the grafeas protos handled occurrence details, but it does keep the proto definitions more DRY than otherwise.

https://github.com/grafeas/grafeas/blob/ff616c8055fd74ae5f14f45e051298584a4e920e/proto/v1beta1/grafeas.proto#L231-L257

Reply:
Meant to include it, but I don't think that's an option if we want to produce similar JSON to the SLSA spec. There will be a nesting with the `oneof` field and the options can't share the same name (`predicate`).